src/services/services.ts

```diff
@@ -5762,6 +5773,62 @@ namespace ts {
             return forEach(diagnostics, diagnostic => diagnostic.category === DiagnosticCategory.Error);
         }
 
+        function getDependencies(fileName: string): DependencyInfo {
+            synchronizeHostData();
+            
+            let sourceFile = getValidSourceFile(fileName);
+            if (!sourceFile) {
+                return undefined;
+            }
+            
+            let resolver = program.getDiagnosticsProducingTypeChecker().getEmitResolver(sourceFile);
+            
+            let runtime: string[] = [];  
+            let compileTime: string[] = [];  
+            for (let node of sourceFile.amdDependencies) {  
+                runtime.push(node.path);  
+            }  
+            
+            function getModuleNameText(node: ImportDeclaration | ImportEqualsDeclaration |  ExportDeclaration): string {
+                let moduleName = getExternalModuleName(node);

```

Comment:
The name is not sufficient. We are making some changes to the resolution logic to honor node resolution process, e.g. looking under node_moduels, and looking in package.json. so you want to leverage that here, so this should be a tuple of name (possibly for error reporting) and resolved name. @vladima has a change for this in #3616.